_posts/2021-11-29-when-estimating-a-software-project-you-should-not.md

```diff
@@ -0,0 +1,34 @@
+---
+title: When estimating a software project you should not...
+---
+
+I'm currently working on an internal GitHub project where, for a change, my estimations have been wildly off. Before this project, I'd had a good multi-year streak of never overshooting my estimates and consistently hitting deadlines. This break down, combined with some recent reading, has made me think about the myths we tell ourselves about software estimation so I figure I'd repeat some here. These are in a rough order from "most obvious" to "least obvious".
+
+When estimating a software project you should not:
+
+- refuse to estimate because of any of the reasons below
+- plan things to go as you expect
+- ignore testing, code review, deployment and integration takes time, often longer than writing the code
+- present exact dates or times without ranges
+- be pressured into different estimates for the same scope based on "business needs"
+- be unwilling to pad estimates based on them "feeling" too long
+- estimate the entire project without splitting it up individually ranged, estimated sub-projects
+- have a non-engineer produce the estimate
+- have an engineer who will not be doing the work produce the estimate, without discussion with the engineer that will be doing the work
+- assume that engineers are will do the work in the same amount of time, even if they are the same "level". [Individuals matter](https://danluu.com/people-matter/).
```

Comment:
```suggestion
- assume that engineers will do the work in the same amount of time, even if they are the same "level". [Individuals matter](https://danluu.com/people-matter/).
```

Reply:
Actually, this might need a bit more fixing, should it be "even if they are _not_ the same level" ?

Reply:
Yeh, rewording to "even if they have the same title" which I think better signals my intent.